staging/src/k8s.io/apimachinery/pkg/api/meta/help.go

```diff
@@ -202,8 +264,56 @@ func ExtractList(obj runtime.Object) ([]runtime.Object, error) {
 	return list, nil
 }
 
-// objectSliceType is the type of a slice of Objects
-var objectSliceType = reflect.TypeOf([]runtime.Object{})
+// ExtractListWithAlloc returns obj's Items element as an array of runtime.Objects.
+// the difference from ExtractList is that an object of the same type is regenerated and assigned a value,
+// so that the returned Object list and obj have independent life cycles.
+// Returns an error if obj is not a List type (does not have an Items member).
+func ExtractListWithAlloc(obj runtime.Object) ([]runtime.Object, error) {
```

Comment:
similar comment, if you could somehow extract the common parts of this pair of functions it would be easier to read.

Reply:
thanks for the review suggestion, it has been modified. PTAL...